dftimewolf/lib/args_validator.py

```diff
@@ -0,0 +1,226 @@
+"""Validators for recipe arguments."""
+
+import abc
+import ipaddress
+import re
+
+from typing import Any, Dict, Optional
+
+from dftimewolf.lib import errors
+
+
+class AbstractValidator(abc.ABC):
+  """Base class for validator objects."""
+
+  name: str = None  # type: ignore
+
+  @abc.abstractmethod
+  def __init__(self) -> None:
+    """Initialise."""
+
+  @abc.abstractmethod
+  def Validate(
+      self, operand: Any, validator_params: Optional[Dict[str, Any]]) -> None:
+    """Validate the parameter.
+
+    Subclasses that override this should raise errors.RecipeArgsValidatorError
+    on validation failures, with a string describing the problem."""
```

Comment:
A description of the parameters here would be nice.

Reply:
Done.